tests_config/run_spec.js

```diff
@@ -113,89 +122,71 @@ global.run_spec = (fixtures, parsers, options) => {
   }
 
   const stringifiedOptions = stringifyOptionsForTitle(options);
+  const [parser, ...verifyParsers] = parsers;
+
+  if (parsers.includes("typescript") && !parsers.includes("babel-ts")) {
+    verifyParsers.push("babel-ts");
+  }
 
   for (const { name, filename, code, output } of [...files, ...snippets]) {
     describe(`${name}${
       stringifiedOptions ? ` - ${stringifiedOptions}` : ""
     }`, () => {
-      let rangeStart;
-      let rangeEnd;
-      let cursorOffset;
-
-      const source = (TEST_CRLF ? code.replace(/\n/g, "\r\n") : code)
-        .replace(RANGE_START_PLACEHOLDER, (match, offset) => {
-          rangeStart = offset;
-          return "";
-        })
-        .replace(RANGE_END_PLACEHOLDER, (match, offset) => {
-          rangeEnd = offset;
-          return "";
-        });
-
-      const input = source.replace(CURSOR_PLACEHOLDER, (match, offset) => {
-        cursorOffset = offset;
-        return "";
-      });
-
-      const baseOptions = {
+      const formatOptions = {
         printWidth: 80,
         ...options,
-        rangeStart,
-        rangeEnd,
-        cursorOffset,
-      };
-      const mainOptions = {
-        ...baseOptions,
-        ...(IS_PARSER_INFERENCE_TESTS
-          ? { filepath: filename }
-          : { parser: parsers[0] }),
+        filepath: filename,
+        parser,
       };
 
-      const hasEndOfLine = "endOfLine" in mainOptions;
-
       if (IS_ERROR_TESTS) {
         test("error test", () => {
           expect(() => {
-            format(input, filename, mainOptions);
+            format(code, formatOptions);
           }).toThrowErrorMatchingSnapshot();
         });
         return;
       }
 
-      const formattedWithCursor = format(input, filename, mainOptions);
-      const formatted = formattedWithCursor.replace(CURSOR_PLACEHOLDER, "");
-      const visualizedOutput = visualizeEndOfLine(formattedWithCursor);
+      const eolReplacedCode = TEST_CRLF ? code.replace(/\n/g, "\r\n") : code;
+      const firstFormatResult = format(eolReplacedCode, formatOptions);
 
       test("format", () => {
-        expect(visualizedOutput).toEqual(
-          visualizeEndOfLine(consistentEndOfLine(formattedWithCursor))
+        // Make sure output has consistent EOL
+        expect(firstFormatResult.eolVisualizedOutput).toEqual(
+          visualizeEndOfLine(
+            consistentEndOfLine(firstFormatResult.outputWithCursor)
+          )
         );
+
         if (typeof output === "string") {
-          expect(formatted).toEqual(output);
+          expect(firstFormatResult.output).toEqual(output);
         } else {
```

Comment:
I'll change this to `early return` before merge, in order to reduce indention of rest part.